src/work/ConditionalWork.cpp

```diff
@@ -97,5 +103,22 @@ void
 ConditionalWork::onReset()
 {
     mWorkStarted = false;
+    mConditionalWorkStartTime.reset();
+}
+
+std::string
+ConditionalWork::getStatus() const
+{
+    std::chrono::milliseconds elapsed{0};
```

Comment:
Are you saying we should use `LogSlowExecution` just for the elapsed time? This function doesn't log anything so that might be confusing. I could use `LogSlowExecution` and set the `threshold` high so nothing gets logged, but I wanted to make sure this is what you meant.